src/components/TodosFilter/TodosFilter.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { useState } from 'react';
+import classNames from 'classnames';
+
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todos: Todo[],
+  setTodos: (a: Todo[]) => void;
+  allTodos: Todo[];
+};
+
+export const TodosFilter: React.FC<Props> = ({ todos, setTodos, allTodos }) => {
+  enum Status {
+    all = 'all',
+    active = 'active',
+    completed = 'Completed',
+  }
+
+  const [filterBy, setFilterBy] = useState<Status>(Status.all);
+
+  const getFilteredTodos = async (filter: Status) => {
+    let filteredTodos: Todo[] = [];
+
+    switch (filter) {
+      case Status.active:
+        filteredTodos = allTodos.filter(todo => !todo.completed);
+        break;
+      case Status.completed:
+        filteredTodos = allTodos.filter(todo => todo.completed);
+        break;
+      default:
+        filteredTodos = allTodos;
+        break;
+    }
+
+    setTodos(filteredTodos);
+  };
+
+  const handleSetFilteredTodos = (filter: Status) => {
+    setFilterBy(filter);
+    getFilteredTodos(filter);
+  };
+
+  const completedTodos
+    = allTodos.filter(todo => todo.completed).length;
+
+  const uncompletedTodos
+    = allTodos.filter(todo => !todo.completed).length;
+
+  return (
+    (todos.length > 0) ? (
```

Comment:
```suggestion
    (todos.length > 0) ? (
```
better to move this checking in App component and render or not TodosFilter component